MarkdownToLatex/MarkdownToLatex/LatexRenderer.cs

```diff
@@ -1,7 +1,213 @@
+using System.Collections.Generic;
+using System;
+using System.Text.RegularExpressions;
+using System.IO;
+
 namespace MarkdownToLatex
 {
-    public class LatexRenderer
+
+    /// <summary>a static class containing methods used for rendering the LaTeX document.</summary>
+    public static class LatexRenderer
     {
-        
+        /// <summary>a List containing the LaTeX document line by line.</summary>
+        public static List<string> LatexLines;
+
+        private static byte _inlist;
+
+        private static byte oldInList;
+
+        /// <summary>a byte indicating whether a list is currently rendered.</summary>
+        public static byte InList {get => _inlist; private set {
+            if(value <= 3){
+                _inlist = value;
+            } else {
+                _inlist = 3; 
+            }
+        }}
+
+        private static byte _inquote;
+        private static byte oldInQuote;
+
+        /// <summary>a byte indicating whether a quote is currently rendered.</summary>
+        public static byte InQuote {get => _inquote; private set {
+            if(value <= 3){
+                _inquote = value;
+            } else {
+                _inquote = 3;
+            }
+        }}
+
+        /// <summary>Writes the LaTeX document with a <paramref name="filename"/> into a 'latex' folder at the specified <paramref name="path"/>.</summary>
+        public static void WriteLatexDocument(string path, string filename){
```

Comment:
Das mit dem Dateinamen ist eher kontraproduktiv besonders und auch so hier nicht komplett implementiert. (Momentan müsste ich in `MdToTex` den Dateinamen des Markdowns entfernen und extra mit `.tex` übergeben.)  
Wir haben hier 2 Möglichkeiten:
1. Wir gehen den Schritt zurück und geben immer den Markdownpfad durch
2. Wir schreiben es universell, also bauen gleich die Option eines optionalen Pfades ein und gehen deine Idee weiter

Im Fall von 2.:
``` C#
public static void WriteLatexDocument(string path){
            string latexPath;
            
            if (Path.GetExtension(path) == ".tex") {
                latexPath = path;
            } else if (Path.GetExtension(path) == ".md") {
                latexPath = Path.Combine(Path.GetDirectoryName(path), Path.GetFileNameWithoutExtension(path) + ".tex");
            } else {
                latexPath = Path.Combine(path, Path.GetFileNameWithoutExtension(MdToTex.mdFilePath) + ".tex");
            }

            Directory.CreateDirectory(latexPath);
...
```

Zum `else`, da kannst du im `MdToTex.mdFilePath` auf `internal` umschreiben.  
**Oder** wir machen es uns ganz einfach und sagen im `else`, `"latexDoc_01.tex"` oder sogar schon `"latexDoc_{num:000}.tex"` und danach `num++;` (für wenn wir Support für Verzeichnisse an Docs einbauen sollten, zerstört ja so nix)